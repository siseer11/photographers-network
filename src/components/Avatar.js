import React from "react";
import PropTypes from "prop-types";
import {StarSVG} from "./svg/StarSVG";

export const Avatar = ({ userImageUrl, classes, profileLink, premium }) => (
  <a className="profile-link" href={profileLink}>
    {premium && <div className="premium-flag"><StarSVG classes="gb-icon-medium gb-icon-fill-white"/></div>}
    <img src={userImageUrl} alt="avatar" className={`gb-avatar ${classes}`} />
  </a>
);

Avatar.propTypes = {
  userImageUrl: PropTypes.string.isRequired,
  classes: PropTypes.string,
  profileLink: PropTypes.string
};

Avatar.defaultProps = {
  classes: "gb-avatar-medium",
  profileLink: "#"
};
