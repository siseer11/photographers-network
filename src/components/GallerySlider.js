import React from "react";
import { initSlider } from "../gallerySlider";

let fakeSlidesData = [
  {
    "image-url":
      "https://images.unsplash.com/photo-1470058869958-2a77ade41c02?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ",
    "slide-title": "Lonely Wolf, Sweden",
    "slide-author": "James Flinton"
  },
  {
    "image-url":
      "https://images.unsplash.com/photo-1503836474039-37fb1d7a4ce3?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ",
    "slide-title": "Melting Ice in Arctic Ocean2",
    "slide-author": "John Scurm"
  },
  {
    "image-url":
      "https://images.unsplash.com/photo-1503515235263-c699356bfb17?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ",
    "slide-title": "Frozen Pant",
    "slide-author": "Fiol Nohn"
  },
  {
    "image-url":
      "https://images.unsplash.com/photo-1538831790613-0b65a044c951?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ",
    "slide-title": "Melting Ice in Arctic Ocean4",
    "slide-author": "Sibo Dingo"
  }
];

export default class GallerySlider extends React.Component {
  componentDidMount() {
    const textBottom = fakeSlidesData.map(el => ({
      "slide-title": el["slide-title"],
      "slide-author": el["slide-title"]
    }));
    initSlider(
      this.listOfBubbles,
      this.sliderWrapper,
      this.listOfSlides,
      textBottom,
      this.bottomTextHolder,
      this.carousel1Title,
      this.carousel1Author
    );
  }
  render() {
    return (
      <div className="gb-image-carousel-1">
        <ul
          ref={node => {
            this.sliderWrapper = node;
          }}
          className="carousel-1-images-list"
        >
          {fakeSlidesData.map((el, idx) => (
            <li
              className={`gallery-slide ${idx == 0 ? "active" : ""}`}
              key={idx}
              ref={node => {
                this.listOfSlides
                  ? this.listOfSlides.push(node)
                  : (this.listOfSlides = [node]);
              }}
            >
              <div
                className="carousel-1-image-item"
                style={{ backgroundImage: `url(${el["image-url"]})` }}
              />
            </li>
          ))}
        </ul>
        <div
          ref={node => {
            this.bottomTextHolder = node;
          }}
          className="carousel-1-about-slide"
        >
          <h2
            ref={node => {
              this.carousel1Title = node;
            }}
            className="carousel-1-slide-title gb-text-black gb-title-medium-large-on-desktop"
          >
            {fakeSlidesData[0]["slide-title"]}
          </h2>
          <a
            ref={node => {
              this.carousel1Author = node;
            }}
            className="carousel-1-slide-author gb-text-black-opacity-50 gb-paragraph-small"
            href="#"
          >
            {fakeSlidesData[0]["slide-author"]}
          </a>
        </div>
        <ul className="carousel-1-pagination">
          {fakeSlidesData.map((el, idx) => (
            <li
              key={idx}
              className={`carousel-1-pagination-buble ${
                idx == 0 ? "active" : ""
              }`}
              ref={node => {
                this.listOfBubbles
                  ? this.listOfBubbles.push(node)
                  : (this.listOfBubbles = [node]);
              }}
              data-index={idx}
            />
          ))}
        </ul>
      </div>
    );
  }
}
