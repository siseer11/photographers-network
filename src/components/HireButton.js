import React from "react";
import WithModal from "../RenderProp/WithModal";
import HireMeModal from "../contents/company/private-job/HireMeModal";

export const HireButton = ({
  uid,
  photographerName,
  siggnedInUser,
  photographerData,
  classes
}) => (
  <WithModal closeItemClass="close">
    {({ showModal, closeModalListener }) => (
      <React.Fragment>
        <div className={classes}>
          <h5>Hire me</h5>
        </div>
        {showModal && (
          <HireMeModal
            closeModal={closeModalListener}
            photographerId={uid}
            photographerName={photographerName}
            company={siggnedInUser}
            photographerData={photographerData}
          />
        )}
      </React.Fragment>
    )}
  </WithModal>
);
