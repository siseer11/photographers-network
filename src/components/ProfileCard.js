import React from "react";
import PropTypes from "prop-types";
import {Avatar} from "./Avatar";

export const ProfileCard = ({
                              backgroundImg,
                              profileImageUrl,
                              premium
                            }) => {
  return (
    <div className="gb-card-10-wrapper">
      <div
        className="profile-card"
        style={{backgroundColor: '#002b41'}}
      >
        <div className="card-10-shadow-overlay"/>
        <Avatar
          premium={premium}
          userImageUrl={
            profileImageUrl ||
            "http://cdn.onlinewebfonts.com/svg/img_74993.png"
          }
          classes="profile-avatar gb-avatar-x-large"
        />
      </div>
    </div>
  );
};

ProfileCard.propTypes = {
  backgroundImg: PropTypes.string,
  type: PropTypes.string.isRequired,
  hireable: PropTypes.bool,
  uid: PropTypes.string.isRequired
};
