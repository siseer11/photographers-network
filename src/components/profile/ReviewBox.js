import React from "react";
import {Avatar} from "../label/Avatar";
import {ArrowSVG} from "../svg/ArrowSVG";

export const ReviewBox = ({title, userImageURL, name, quote}) => (
  <div className="review-box">
    <h2>{title}</h2>
    <Avatar userImageUrl={userImageURL}/>
    <h3>John Doe</h3>
    <p>“{quote}”</p>
    <ArrowSVG classes="gb-icon-medium"/>
  </div>
);