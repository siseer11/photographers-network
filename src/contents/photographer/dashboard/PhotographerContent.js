import React from "react";
import { PortofolioGallery } from "../../../components/PortofolioGallery";
import { PropTypes } from "prop-types";
import {HireButton} from "../../../components/HireButton";
import {Link} from "react-router-dom";
import {ReviewBox} from "../../../components/profile/ReviewBox";

export const PhotographerContent = ({ photographerData, isOtherUser, siggnedInUser: signedInUser, uid, finishedJobs, hireable }) => (
  <React.Fragment>
    <div className="flex-container-space">
      <div>
        <span className="big-num">{finishedJobs}</span>
        <span className="light-blue">Jobs <br/>done</span>
      </div>
      <Link
        to="/dashboard"
        className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      >
        See the projects
      </Link>
    </div>
    {photographerData.portfolio && photographerData.portfolio.length > 0 && (
      <React.Fragment>
        <h2 className="black-header-portfolio">Portfolio</h2>
        <PortofolioGallery photosList={photographerData.portfolio} />
      </React.Fragment>
    )}
    {
      hireable &&
      <div className="pink-box">
        <p>There, between rolling hills and the Apennine mountains in the Mugello valley.</p>
        <HireButton
          classes="hire-button"
          photographerData={photographerData}
          siggnedInUser={signedInUser}
          uid={uid}
          photographerName={`${photographerData.firstName} ${photographerData.lastName}`}
        />
      </div>
    }
    <ReviewBox title="Last review"
               userImageURL="https://images.askmen.com/1080x540/2016/01/25-021526-facebook_profile_picture_affects_chances_of_getting_hired.jpg"
               name="John Doe"
               quote="New battery for smartphones can be charged in a minute"
    />
  </React.Fragment>
);

PhotographerContent.propTypes = {
  photographerData: PropTypes.object.isRequired
};
