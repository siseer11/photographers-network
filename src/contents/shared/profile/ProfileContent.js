import React from "react";
import {PhotographerContent} from "../../photographer/dashboard/PhotographerContent";
import {HireButton} from "../../../components/HireButton";
import {Link} from "react-router-dom";
import {PencilSVG} from "../../../components/svg/PencilSVG";

export const ProfileContent = ({thisProfileData, currentUserData, currentUserId, otherUser, finishedJobs}) => (
  <div className="profile-content">
    <h1>{thisProfileData.companyName ||
    `${thisProfileData.firstName} ${thisProfileData.lastName}`}</h1>
    <p>
      <span className="profile-desc bold">{Object.values(thisProfileData.locations)[0].city}</span>
      <span> | </span>
      <span className="profile-desc">{thisProfileData.type === "photographer" ? "Photographer" : "Company"}</span>
    </p>
    {
      (
        thisProfileData.type === "photographer" &&
        currentUserData.type === "company" &&
        thisProfileData.hireable
      ) &&
      <HireButton
        classes="gb-btn gb-btn-small gb-btn-primary gb-btn-white hire"
        photographerData={thisProfileData}
        siggnedInUser={currentUserData}
        uid={currentUserId}
        photographerName={`${thisProfileData.firstName} ${thisProfileData.lastName}`}
      />
    }
    {!otherUser && (
      <Link
        to="/ProfileEdit"
        className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      >
        Edit Profile <PencilSVG classes="gb-icon-medium gb-icon-fill-white"/>
      </Link>
    )}
    <div className="black">
      <h3>Description</h3>
      <p className="profile-desc">
        There, between rolling hills and the Apennine mountains in the Mugello valley (home to the famous race
        track) lies Lake Bilancino.
        There, between rolling hills and the Apennine mountains in the Mugello valley.

        There, between rolling hills and the Apennine mountains in the Mugello valley (home to the famous race
        track) lies Lake Bilancino.
      </p>
    </div>

    {thisProfileData.type === "photographer" ? (
      <PhotographerContent
        photographerData={thisProfileData}
        isOtherUser={otherUser}
        siggnedInUser={currentUserData}
        uid={currentUserId}
        finishedJobs={finishedJobs}
        hireable={thisProfileData.hireable}
      />
    ) : (
      <div>Rest of the company content</div>
    )}
  </div>
);