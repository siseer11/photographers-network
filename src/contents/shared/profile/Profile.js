// dependencies
import React from "react";
import { connect } from "react-redux";
import { compose } from "redux";
import { firestoreConnect, isLoaded, isEmpty } from "react-redux-firebase";

// components
import { ProfileCard } from "../../../components/ProfileCard";
import { ProfileContent } from "./ProfileContent";

export const Profile = ({
  match,
  profileData,
  currentUserData,
  currentUserId,
  finishedJobs
}) => {
  const profileId = match.params.uid;
  if (!isLoaded(profileData)) {
    return <h2 className="loading">Loading..</h2>;
  } else if (profileData && !isLoaded(profileData[profileId])) {
    return <h2 className="loading person-data-not-loaded">Loading...</h2>;
  }

  if (isEmpty(profileData[profileId])) {
    return <h2 className="no-data">No data for this id.</h2>;
  }

  const otherUser = currentUserId !== profileId;
  const thisProfileData = profileData[profileId];

  return (
    <div className="profile">
      <ProfileCard {...thisProfileData} siggnedInUser={currentUserData} />
      <ProfileContent
        thisProfileData={thisProfileData}
        currentUserData={currentUserData}
        currentUserId={currentUserId}
        otherUser={otherUser}
        finishedJobs={finishedJobs}
      />
    </div>
  );
};

const mapStateToProps = state => {
  const jobOffers = state.firestore.ordered.jobOffers;
  return {
    profileData: state.firestore.data.users,
    currentUserData: state.firebase.profile,
    currentUserId: state.firebase.auth.uid,
    finishedJobs: jobOffers ? jobOffers.length : 0
  };
};

export default compose(
  connect(mapStateToProps),
  firestoreConnect(props => {
    return [
      {
        collection: `users`,
        doc: props.match.params.uid
      },
      {
        collection: "jobOffers",
        where: [
          ["photographer.uid", "==", props.match.params.uid],
          ["status", "==", "closed"]
        ]
      }
    ];
  })
)(Profile);
