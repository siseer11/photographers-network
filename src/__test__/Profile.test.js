import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import makeUser from "./mockUser";
import { Link } from "react-router-dom";
import { Profile } from "../contents/shared/profile/Profile";
import { shallow } from "enzyme";
import { ProfileCard } from "../components/ProfileCard";
import { ProfileContent } from "../contents/shared/profile/ProfileContent";
import { Avatar } from "../components/Avatar";
import { PortofolioGallery } from "../components/PortofolioGallery";
import { ReviewBox } from "../components/profile/ReviewBox";
import { PhotographerContent } from "../contents/photographer/dashboard/PhotographerContent";

describe("Profile tests", () => {
  const match = { params: { uid: "fakeid" } }; //the match prop is needed always, so it takes the uid from the link query param
  const shallowRender = props => shallow(<Profile match={match} {...props} />);

  test("Profile it is defined", () => {
    const shallowProfile = shallowRender();
    expect(shallowProfile).toBeDefined();
  });

  describe("it is showing loading correctly", () => {
    const shallowProfile = shallowRender();
    it("shows it if the profileData is not loaded", () => {
      shallowProfile.setProps({ profileData: undefined });
      expect(shallowProfile.find(".loading").length).toBe(1);
    });

    it("shows is if this person data is not loaded", () => {
      const match = { params: { uid: "fakeid" } };
      shallowProfile.setProps({
        profileData: { fakeid: undefined },
        match
      });

      expect(shallowProfile.find(".loading").length).toBe(1);
    });
  });

  it("shows 'no data found' correctly", () => {
    const shallowProfile = shallowRender();
    const match = { params: { uid: "fakeid" } };
    shallowProfile.setProps({
      profileData: { fakeid: null },
      match
    });

    expect(shallowProfile.find(".no-data").length).toBe(1);
  });

  describe("renders correctly with all the props passed", () => {
    //A mock of the props that it will take in a real case;
    let mockProps = {
      match: { params: { uid: "otherUser" } },
      profileData: {
        otherUser: {
          firstName: "z",
          lastName: "z",
          hireable: true,
          locations: {
            "123azE": {
              city: "S",
              country: "Z",
              streetName: "A",
              streetNumber: "83"
            }
          },
          premium: true,
          portfolio: [],
          profileImageUrl: "img.jpg",
          type: "photographer",
          uid: "otherUser"
        }
      },
      currentUserData: {
        firstName: "z",
        lastName: "z",
        hireable: true,
        locations: {
          "123azE": {
            city: "S",
            country: "Z",
            streetName: "A",
            streetNumber: "83"
          }
        },
        premium: true,
        portfolio: [],
        profileImageUrl: "img.jpg",
        type: "photographer",
        uid: "fakeid"
      },
      currentUserId: "fakeid",
      finishedJobs: 0
    };
    let shallowProfile = shallowRender(mockProps);

    it("renders once a ProfileCard", () => {
      expect(shallowProfile.find(ProfileCard).length).toBe(1);
    });

    it("renders once a ProfileContent", () => {
      expect(shallowProfile.find(ProfileContent).length).toBe(1);
    });

    it("pass down correct otherUser prop", () => {
      expect(
        shallowProfile.find(ProfileContent).get(0).props.otherUser
      ).toEqual(true);
    });
  });
});

describe("ProfileCard", () => {
  it("renders corectly", () => {
    const tree = renderer
      .create(
        <ProfileCard profileImageUrl={"myFakeImage.jpg"} premium={false} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Avatar", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Avatar
          userImageUrl="image.jpg"
          classes="premium user-avatar"
          profileLink="user1"
          premium={true}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("does not show premium mark for users that are not premium", () => {
    const shallowAvatar = shallow(
      <Avatar
        userImageUrl="image.jpg"
        classes="premium user-avatar"
        profileLink="user1"
        premium={false}
      />
    );

    expect(shallowAvatar.find(".premium-flag").length).toEqual(0);
  });

  it("does show premium mark for premium users", () => {
    const shallowAvatar = shallow(
      <Avatar
        userImageUrl="image.jpg"
        classes="premium user-avatar"
        profileLink="user1"
        premium={true}
      />
    );

    expect(shallowAvatar.find(".premium-flag").length).toEqual(1);
  });
});

describe("ProfileContent", () => {
  let currentUserData = makeUser({ type: "photographer" });

  let currentUserId = currentUserData.uid;
  let otherUser = false;
  let finishedJobs = 0;
  let thisProfileData = makeUser({ type: "photographer", uid: "aaa" });
  const shallowRender = props =>
    shallow(
      <ProfileContent
        thisProfileData={thisProfileData}
        currentUserData={currentUserData}
        currentUserId={currentUserId}
        otherUser={otherUser}
        finishedJobs={finishedJobs}
        {...props}
      />
    );

  it("renders corectly", () => {
    const tree = renderer
      .create(
        <Router>
          <ProfileContent
            thisProfileData={thisProfileData}
            currentUserData={currentUserData}
            currentUserId={currentUserId}
            otherUser={otherUser}
            finishedJobs={finishedJobs}
          />
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("renders correctly the profile name", () => {
    it("shows correctly for photographers", () => {
      const shallowElPhotographer = shallowRender();
      expect(shallowElPhotographer.find(".profile-name").text()).toBe(
        thisProfileData.firstName + " " + thisProfileData.lastName
      );
    });

    it("shows correctly for companies", () => {
      const companyData = {
        ...currentUserData,
        type: "company",
        companyName: "Z.co",
        firstName: null,
        lastName: null
      };
      const shallowElCompany = shallowRender({
        thisProfileData: companyData
      });
      expect(shallowElCompany.find(".profile-name").text()).toBe(
        companyData.companyName
      );
    });
  });

  describe("shows correctly the hireButton", () => {
    let currentUserData = makeUser({ type: "company" });

    it("shows it if all conditions are filled", () => {
      const photographerData = makeUser({
        type: "photographer",
        hireable: true
      });
      const shallowElement = shallowRender({
        thisProfileData: photographerData,
        currentUserData: currentUserData
      });

      expect(shallowElement.find("HireButton").length).toEqual(1);
    });

    describe("does not show it if", () => {
      it("is not a photographer profile", () => {
        const thisProfileData = makeUser({
          type: "company"
        });
        const shallowElement = shallowRender({
          thisProfileData: thisProfileData,
          currentUserData: currentUserData
        });

        expect(shallowElement.find("HireButton").length).toEqual(0);
      });

      it("current user is not a company", () => {
        currentUserData = makeUser({ type: "photographer" });

        const thisProfileData = makeUser({
          type: "photographer"
        });

        const shallowElement = shallowRender({
          thisProfileData: thisProfileData,
          currentUserData: currentUserData
        });

        expect(shallowElement.find("HireButton").length).toEqual(0);
      });

      it("this profile user is a photographer but has hireable=false", () => {
        thisProfileData = makeUser({ type: "photographer", hireable: false });

        const shallowElement = shallowRender({
          thisProfileData: thisProfileData,
          currentUserData: currentUserData
        });

        expect(shallowElement.find("HireButton").length).toEqual(0);
      });
    });
  });

  describe("shows correctly the edit profile button", () => {
    const shallowElement = shallowRender();

    it("does not show it, if is another user profile", () => {
      shallowElement.setProps({ otherUser: true });
      expect(shallowElement.find("#edit-profile-link").length).toEqual(0);
    });

    it("does show it, if it is your own profile", () => {
      shallowElement.setProps({ otherUser: false });
      expect(shallowElement.find("#edit-profile-link").length).toEqual(1);
    });
  });

  describe("shows correctly aditional informations for company/photographer", () => {
    it("shows correctly for photographer profile", () => {
      const thisProfileData = makeUser({ type: "photographer" });
      const shallowElement = shallowRender({
        thisProfileData: thisProfileData
      });

      expect(shallowElement.find("PhotographerContent").length).toEqual(1);
    });

    it("shows correctly for company profile", () => {
      const thisProfileData = makeUser({ type: "company" });
      const shallowElement = shallowRender({
        thisProfileData: thisProfileData
      });

      expect(shallowElement.find(".company-details").length).toEqual(1);
      expect(shallowElement.find("PhotographerContent").length).toEqual(0);
    });
  });
});

describe("PhotographerContent", () => {
  const photographerData = makeUser({ type: "photographer" });
  const siggnedInUser = makeUser({ type: "company" });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <PhotographerContent
            photographerData={photographerData}
            siggnedInUser={siggnedInUser}
            uid={photographerData.uid}
            finishedJobs={photographerData.finishedJobs}
            hireable={photographerData.hireable}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const shallowElement = shallow(
    <PhotographerContent
      photographerData={photographerData}
      siggnedInUser={siggnedInUser}
      uid={photographerData.uid}
      finishedJobs={photographerData.finishedJobs}
      hireable={photographerData.hireable}
    />
  );

  it("renders the correct number of finishedJobs", () => {
    expect(shallowElement.find(".flex-container-space .big-num").text()).toBe(
      "0"
    );

    shallowElement.setProps({ finishedJobs: 10 });

    expect(shallowElement.find(".flex-container-space .big-num").text()).toBe(
      "10"
    );
  });

  it("renders the link to project once", () => {
    expect(
      shallowElement.containsMatchingElement(
        <Link to="/dashboard">See the projects</Link>
      )
    ).toBe(true);
  });

  describe("renders the portofolio correctly, ", () => {
    it("if the photographer have no portfolio data", () => {
      const newPhotographerData = makeUser({
        type: "photographer",
        portfolio: null
      });
      shallowElement.setProps({ photographerData: newPhotographerData });

      expect(shallowElement.find("PortofolioGallery").length).toBe(0);
    });

    it("shows it when photographer have a portfolio", () => {
      const newPhotographerData = makeUser({
        type: "photographer",
        portfolio: []
      });
      shallowElement.setProps({ photographerData: newPhotographerData });

      expect(shallowElement.find("PortofolioGallery").length).toBe(0);
    });

    it("shows nothing if the portfolio is empty", () => {
      const newPhotographerData = makeUser({
        type: "photographer",
        portfolio: ["img1.jpg", "img2.jpg"]
      });
      shallowElement.setProps({ photographerData: newPhotographerData });

      expect(shallowElement.find("PortofolioGallery").length).toBe(1);
    });
  });

  describe("renders the hire button correctly", () => {
    it("shows it , if photographer is hireable and the currentUser is a company", () => {
      const siggnedInUser = makeUser({ type: "company" });
      const photographer = makeUser({ type: "photographer", hireable: true });

      shallowElement.setProps({
        photographerData: photographer,
        siggnedInUser: siggnedInUser,
        hireable: true
      });

      expect(shallowElement.find("HireButton").length).toBe(1);
    });

    it("does not show it , if the photographer is not hireable", () => {
      const siggnedInUser = makeUser({ type: "company" });
      const photographer = makeUser({ type: "photographer", hireable: false });

      shallowElement.setProps({
        photographerData: photographer,
        siggnedInUser: siggnedInUser,
        hireable: false
      });

      expect(shallowElement.find("HireButton").length).toBe(0);
    });

    it("does not show it , if the currentUser is not a company", () => {
      const siggnedInUser = makeUser({ type: "photographer" });
      const photographer = makeUser({ type: "photographer", hireable: true });

      shallowElement.setProps({
        photographerData: photographer,
        siggnedInUser: siggnedInUser,
        hireable: true
      });

      expect(shallowElement.find("HireButton").length).toBe(0);
    });
  });

  it("always renders a ReviewBox", () => {
    expect(shallowElement.find("ReviewBox").length).toBe(1);
  });
});
