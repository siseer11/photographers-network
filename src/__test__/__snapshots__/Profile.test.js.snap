// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<a
  className="profile-link"
  href="user1"
>
  <div
    className="premium-flag"
  >
    <svg
      className="gb-icon-medium gb-icon-fill-white"
      height="100%"
      id="facebook"
      viewBox="0 0 28 28"
      width="100%"
    >
      <title>
        search2
      </title>
      <path
        d="M13.843,5.323 L15.832,9.807 C16.164,10.558 16.841,11.101 17.646,11.262 L23.302,12.396 L19.423,16.275 C18.827,16.871 18.566,17.726 18.728,18.555 L19.811,24.104 L14.868,21.908 C14.542,21.763 14.193,21.690 13.843,21.690 C13.494,21.690 13.144,21.763 12.818,21.908 L7.872,24.105 L8.956,18.554 C9.117,17.726 8.857,16.872 8.261,16.276 L4.385,12.395 L10.040,11.262 C10.845,11.101 11.521,10.558 11.855,9.807 L13.843,5.323 M13.843,0.650 C13.421,0.650 13.038,0.901 12.865,1.287 L9.544,8.777 L1.210,10.449 C0.825,10.528 0.515,10.811 0.401,11.190 C0.285,11.565 0.389,11.974 0.669,12.257 L6.473,18.068 L4.852,26.388 C4.771,26.794 4.875,27.188 5.189,27.441 C5.376,27.588 5.616,27.670 5.866,27.670 C6.015,27.670 6.163,27.638 6.295,27.577 L13.843,24.224 L21.403,27.583 C21.531,27.644 21.674,27.675 21.827,27.675 C22.068,27.675 22.304,27.594 22.490,27.446 C22.806,27.195 22.911,26.789 22.835,26.390 L21.209,18.068 L27.018,12.257 C27.298,11.978 27.402,11.568 27.287,11.193 C27.173,10.813 26.860,10.528 26.474,10.449 L18.143,8.777 L14.820,1.285 C14.649,0.900 14.265,0.650 13.843,0.650 L13.843,0.650 Z"
        fill="rgb(255, 255, 255)"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <img
    alt="avatar"
    className="gb-avatar premium user-avatar"
    src="image.jpg"
  />
</a>
`;

exports[`PhotographerContent renders correctly 1`] = `
Array [
  <div
    className="flex-container-space"
  >
    <div>
      <span
        className="big-num"
      >
        0
      </span>
      <span
        className="light-blue"
      >
        Jobs 
        <br />
        done
      </span>
    </div>
    <a
      className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      href="/dashboard"
      onClick={[Function]}
    >
      See the projects
    </a>
  </div>,
  <div
    className="pink-box"
  >
    <p>
      There, between rolling hills and the Apennine mountains in the Mugello valley.
    </p>
    <div
      onClick={[Function]}
    >
      <div
        className="hire-button"
      >
        <h5>
          Hire me
        </h5>
      </div>
    </div>
  </div>,
  <div
    className="review-box"
  >
    <h2
      className="review-box-title"
    >
      Last review
    </h2>
    <a
      className="profile-link"
      href="#"
    >
      <img
        alt="avatar"
        className="gb-avatar gb-avatar-medium"
        src="https://images.askmen.com/1080x540/2016/01/25-021526-facebook_profile_picture_affects_chances_of_getting_hired.jpg"
      />
    </a>
    <h3
      className="review-author-name"
    >
      John Doe
    </h3>
    <p
      className="review-author-quote"
    >
      “
      New battery for smartphones can be charged in a minute
      ”
    </p>
    <svg
      className="gb-icon-medium"
      height="100%"
      id="notification"
      viewBox="0 0 185 362"
      width="100%"
    >
      <title>
        arrow
      </title>
      <line
        stroke="grey"
        strokeWidth="10px"
        x1="0.346633"
        x2="184.347"
        y1="0.639659"
        y2="177.64"
      />
      <line
        stroke="grey"
        strokeWidth="10px"
        x1="184.353"
        x2="0.352587"
        y1="178.355"
        y2="361.355"
      />
    </svg>
  </div>,
]
`;

exports[`ProfileCard renders corectly 1`] = `
<div
  className="gb-card-10-wrapper"
>
  <div
    className="profile-card"
    style={
      Object {
        "backgroundColor": "#002b41",
      }
    }
  >
    <div
      className="card-10-shadow-overlay"
    />
    <a
      className="profile-link"
      href="#"
    >
      <img
        alt="avatar"
        className="gb-avatar profile-avatar gb-avatar-x-large"
        src="myFakeImage.jpg"
      />
    </a>
  </div>
</div>
`;

exports[`ProfileContent renders corectly 1`] = `
<div
  className="profile-content"
>
  <h1
    className="profile-name"
  >
    S S
  </h1>
  <p>
    <span
      className="profile-desc bold"
    >
      S
    </span>
    <span>
       | 
    </span>
    <span
      className="profile-desc user-type"
    >
      photographer
    </span>
  </p>
  <a
    className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
    href="/ProfileEdit"
    id="edit-profile-link"
    onClick={[Function]}
  >
    Edit Profile 
    <svg
      className="gb-icon-medium gb-icon-fill-white"
      height="100%"
      id="edit"
      viewBox="0 0 32 32"
      width="100%"
    >
      <title>
        edit
      </title>
      <path
        d="M27 0c2.761 0 5 2.239 5 5 0 1.126-0.372 2.164-1 3l-2 2-7-7 2-2c0.836-0.628 1.874-1 3-1zM2 23l-2 9 9-2 18.5-18.5-7-7-18.5 18.5zM22.362 11.362l-14 14-1.724-1.724 14-14 1.724 1.724z"
      />
    </svg>
  </a>
  <div
    className="black"
  >
    <h3>
      Description
    </h3>
    <p
      className="profile-desc"
    >
      There, between rolling hills and the Apennine mountains in the Mugello valley (home to the famous race track) lies Lake Bilancino. There, between rolling hills and the Apennine mountains in the Mugello valley. There, between rolling hills and the Apennine mountains in the Mugello valley (home to the famous race track) lies Lake Bilancino.
    </p>
  </div>
  <div
    className="flex-container-space"
  >
    <div>
      <span
        className="big-num"
      >
        0
      </span>
      <span
        className="light-blue"
      >
        Jobs 
        <br />
        done
      </span>
    </div>
    <a
      className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      href="/dashboard"
      onClick={[Function]}
    >
      See the projects
    </a>
  </div>
  <div
    className="review-box"
  >
    <h2
      className="review-box-title"
    >
      Last review
    </h2>
    <a
      className="profile-link"
      href="#"
    >
      <img
        alt="avatar"
        className="gb-avatar gb-avatar-medium"
        src="https://images.askmen.com/1080x540/2016/01/25-021526-facebook_profile_picture_affects_chances_of_getting_hired.jpg"
      />
    </a>
    <h3
      className="review-author-name"
    >
      John Doe
    </h3>
    <p
      className="review-author-quote"
    >
      “
      New battery for smartphones can be charged in a minute
      ”
    </p>
    <svg
      className="gb-icon-medium"
      height="100%"
      id="notification"
      viewBox="0 0 185 362"
      width="100%"
    >
      <title>
        arrow
      </title>
      <line
        stroke="grey"
        strokeWidth="10px"
        x1="0.346633"
        x2="184.347"
        y1="0.639659"
        y2="177.64"
      />
      <line
        stroke="grey"
        strokeWidth="10px"
        x1="184.353"
        x2="0.352587"
        y1="178.355"
        y2="361.355"
      />
    </svg>
  </div>
</div>
`;
